libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/HttpAccess.java

```diff
@@ -176,7 +176,7 @@ public void onReceive(Context context, Intent intent) {
 
         // Tries mobile connection.
         final NetworkInfo mobileInfo = conMgr.getNetworkInfo(ConnectivityManager.TYPE_MOBILE);
-        if (mobileInfo == null) {
+        if (mobileInfo == null || mobileInfo.getDetailedState() == NetworkInfo.DetailedState.DISCONNECTED) {
```

Comment:
We should probably use the same pattern as above, and change this block as follows:

```
    if (mobileInfo != null && mobileInfo.isConnected()) {
        setHasNetwork(true);
        return;
    }
    setHasNetwork(false);
    return;
}
```